### DISCUSSION TITLE
Stream from Smartphone (Android) to Chromecast

### DISCUSSION THREAD

**Discussion by mgogoulos** [mediacms-io/mediacms#590] [Feature Requests]
### Discussed in https://github.com/mediacms-io/mediacms/discussions/421

<div type='discussions-op-text'>

<sup>Originally posted by **streiman** February 22, 2022</sup>
I just installed Mediacms on my homeserver for testing - and I am really impressed. But I have a problem / question. When I watch a video on my smartphone (Android, Pixel 6, Chrome Browser) I get the button to stream the video to my chromecast on my TV. But when I select it, the TV only shows the "Chrome" logo, no video is playing.

Is this a normal behaviour or is there anything I can do to enable casting to my TV?

If I stream from Chrome on my notebook everything works fine.</div>

**Comment by mgogoulos** [comment]
Potentially by adding a video.js plugin as https://github.com/silvermine/videojs-chromecast

**Comment by streiman** [comment]
> Potentially by adding a video.js plugin as https://github.com/silvermine/videojs-chromecast

Sorry for the late response. I played around with some different solutions, but MediaCMS is looking the best.

I read the link you posted, but I am running MediaCMS using docker-compose and even I am not unexperienced  I am unsure how to achieve it with docker.

**Comment by morbificagent** [comment]
Hi together,

would love to see this too but for me as a "cant-code-guy" the implemantation of https://github.com/silvermine/videojs-chromecast looks impossible without destroying everything ;-)

Anyone here who can do this?

**Comment by morbificagent** [comment]
Made it... was a hard way for me as i cant code and have to try most of the time, but its working.
I think someone with more experience could implement it nicer but here it is:

1. i am using the files from https://github.com/silvermine/videojs-chromecast via "npm install @silvermine/videojs-chromecast" as the other ways of installing it gave me only errors.

3. Copied the files from the downloaded "node_modules/@silvermine/videojs-chromecast/dist/*" into /home/mediacms.io/mediacms/static/lib/chromecast/ and gave propper rights (www-data)

4. in /home/mediacms.io/mediacms/templates/root.html added some scripts and css:
<img width="892" alt="Bildschirmfoto 2023-06-23 um 23 30 36" src="https://github.com/mediacms-io/mediacms/assets/50199132/bf76af66-6c5d-49b4-a445-03f920d63201">

5. in /home/mediacms.io/mediacms/frontend/src/static/js/components/media-viewer/VideoViewer/index.js added the chromecast-plugin initialisation:
<img width="421" alt="Bildschirmfoto 2023-06-23 um 23 36 24" src="https://github.com/mediacms-io/mediacms/assets/50199132/eda6b051-2424-4360-9b59-5185a1822910">

6. in /home/mediacms.io/mediacms/frontend execute "npm run dist" and after that has finished "cp -rf /home/mediacms.io/mediacms/frontend/dist/static /home/mediacms.io/mediacms/" to copy the changed files... and "systemctl restart mediacms" for sure...

Ok, i think i havent missed something... Result:

![Screenshot_20230623_235011_Kiwi Browser.jpg](https://github.com/mediacms-io/mediacms/assets/50199132/f52517b9-6e2c-4281-8b71-429922aac36c)

I think I will change the scaling of the icon but hey it's working / casting 😉👍
